### PR TITLE
B feature/request leave

### PR DESCRIPTION
**Pull request by dujfalusi**

### CODE REVIEW

File: src/main/java/eu/execom/hawaii/service/AbsenceService.java [hunk 1]
```diff
@@ -63,4 +67,10 @@ public void delete(Long id) {
     absenceRepository.deleteById(id);
   }
 
+  private void checkIsAbsenceLeave(Absence absence) {
+    if (absence.getAbsenceType().equals(AbsenceType.LEAVE)) {
```
Comment (sugrinov) [inline src/main/java/eu/execom/hawaii/service/AbsenceService.java]: What if `absenceType` is null? Maybe it cannot be `null` in this case but it is best practice to always check against a literal (`ENUM`, `constants`, etc) rather then the field itself `AbsenceType.LEAVE.equals(absence.getAbsenceType())`

Comment (sugrinov) [reply src/main/java/eu/execom/hawaii/service/AbsenceService.java]: Also what do you think about moving the comparison directly into the setter?

Comment (luinnarn) [reply src/main/java/eu/execom/hawaii/service/AbsenceService.java]: Another option is a method in the Absence class:
`boolean isLeave() { return AbsenceType.LEAVE.equals(this.absenceType) }`

File: src/main/java/eu/execom/hawaii/service/AbsenceService.java [hunk 2]
```diff
@@ -63,4 +67,10 @@ public void delete(Long id) {
...
+      absence.setDeducted(true);
+    }
```
Comment (luinnarn) [inline src/main/java/eu/execom/hawaii/service/AbsenceService.java]: Not all absences of type `LEAVE` are deducted...

File: src/main/java/eu/execom/hawaii/service/RequestService.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  public Request save(Request request) {
+    checkUserWithApprover(request);
+    return requestRepository.save(request);
+  }
+
+  private void checkUserWithApprover(Request request) {
+    if (request.getUser().getId().equals(request.getApprover().getId())) {
+      throw new ApproverException();
+    }
+  }
```
Comment (luinnarn) [inline src/main/java/eu/execom/hawaii/service/RequestService.java]: Can this situation actually ever happen? Isn't an approverId set only after one of the possible approvers for the user's team actually approves the request? A team can have multiple approvers, none of which can belong to the given team, which effectively aliminates such a possibility...

**Pull request merged**